athena-storage-api/src/main/java/com/amazonaws/athena/storage/datasource/CsvDatasource.java

```diff
@@ -0,0 +1,342 @@
+                            Split split, BlockSpiller spiller, QueryStatusChecker queryStatusChecker)
+    {
+        String databaseName = tableInfo.getSchemaName();
+        String tableName = tableInfo.getTableName();
+        if (!storeCheckingComplete) {
+            this.checkMetastoreForAll(databaseName);
+        }
+        String bucketName;
+        List<String> objectNames = null;
+        bucketName = databaseBuckets.get(databaseName);
+        if (bucketName == null) {
+            throw new DatabaseNotFoundException("No schema '" + databaseName + "' found");
+        }
+        Map<String, List<String>> tableObjectMap = tableObjects.get(databaseName);
+        if (tableObjectMap != null) {
+            objectNames = tableObjectMap.get(tableName);
+        }
+        if (objectNames == null) {
+            throw new TableNotFoundException("No table '" + tableName + "' found under schema '" + databaseName + "'");
+        }
+        try {
+            readRecordsForRequest(schema, constraints, tableInfo, split, storage, bucketName, spiller, queryStatusChecker);
+        }
+        catch (Exception exception) {
+            throw new UncheckedStorageDatasourceException("Error occurred during reading CSV records. Table name : " + tableName
+                    + ", schema name: " + tableInfo.getSchemaName() + ", original file name(s): " + objectNames
+                    + ", bucket name: " +  bucketName + ". Error message: "
+                    + exception.getMessage(), exception);
+        }
+    }
+
+    /**
+     * Return a list of Field instances with field name and field type (Arrow type)
+     *
+     * @param bucketName  Name of the bucket
+     * @param objectNames Name of the file in the specified bucket
+     * @return List of field instances
+     * @throws IOException Raises if any occurs
+     */
+    @Override
+    protected List<Field> getTableFields(String bucketName, List<String> objectNames) throws IOException
+    {
+        try {
+            requireNonNull(objectNames, "List of tables in bucket " + bucketName + " was null");
+            if (objectNames.isEmpty()) {
+                throw new UncheckedStorageDatasourceException("List of tables in bucket " + bucketName + " was empty");
+            }
+            ImmutableList.Builder<Field> fieldListBuilder = ImmutableList.builder();
+            List<String> fieldNames = inferSchemaFields(bucketName, objectNames.get(0));
+            for (String field : fieldNames) {
+                fieldListBuilder.add(new Field(field.toLowerCase(),
+                        FieldType.nullable(new ArrowType.Utf8()), null));
+            }
+            return fieldListBuilder.build();
+        }
+        catch (Exception exception) {
+            LOGGER.error("Unable to retrieve field schema for file(s) {}, under the bucket {}", objectNames,
+                    bucketName);
+            throw new UncheckedStorageDatasourceException(exception.getMessage(), exception);
+        }
```

Comment:
Same comment here, instead of catching to log and then rethrow, just log something saying that you're starting some operation + everything you want before you start the block of code that may throw.

Reply:
This may help quickly spot the issue and fix it since the user can report from which files(s) and bucket it raised the error from the Athena Console.

Reply:
The user still has to look at the logs for the issue. I'm just saying to log before you start the operation with the same information as your current log.error except it will just be a log.info about the retrieval starting.
To be explicit, replace line 207 with this:
```
LOGGER.info("Attempting to retrieve field schema for file(s) {}, under, the bucket {}", objectName, bucketName);
```
And then just delete the catch block.

It will be just as quick for the user because if you don't catch the exception, it will just bubble all the way up and then they will see the exception in the log and then just look up one line to see the log.info message about the retrieval starting and it will contain the same info about the object name and bucket name.

Reply:
try/catch block has been removed accordingly